analysis/customrules.yaml

```diff
@@ -27,30 +27,54 @@ falco:
     maxBurst: 10000
   webserver:
     enabled: false
+extraArgs:
+  - "-A"  # https://falco.org/docs/rules/#ignored-system-calls
 customRules:
   rules-custom.yaml: |-
     - list: allowed_files
       items: [/etc/ld.so.cache, /etc/hosts, /usr/lib/ssl/openssl.cnf, /etc/nsswitch.conf, /etc/resolv.conf]
+    - macro: stat_file
+      condition: evt.type in (access, stat, stat64, lstat, lstat64)
+    - macro: open_file
+      condition: (evt.type in (open, openat) and evt.is_open_read=true)
+    - rule: Unexpected file stat
+      desc: Detect file stat outside the workingdir
+      condition: >
+        (stat_file and
+         not evt.arg.path startswith /app and
+         not evt.arg.path startswith /tmp and
+         not evt.arg.path startswith /usr/local and
+         not evt.arg.path startswith /usr/lib and
+         not evt.arg.path startswith /usr/share and
+         not evt.arg.path startswith /lib and
+         not evt.arg.path startswith /proc and
```

Comment:
I wonder if we can be more precise here. /proc can be used to glean information about processes... say, /proc/<pic>/mem and others.

Reply:
Yes, if we get to that point, it'll likely become a cat and mouse game that we need to address to avoid fingerprinting. We need to get to that point first though :)

Reply:
(And note these don't actually prevent access, these just mean we don't log it)